korman/operators/op_mesh.py

```diff
@@ -387,6 +387,120 @@ def origin_to_bottom(obj):
     mw.translation = global_origin
 
 
+class PlasmaAddLinkingBookMeshOperator(PlasmaMeshOperator, bpy.types.Operator):
+    bl_idname = "mesh.plasma_linkingbook_add"
+    bl_label = "Add Linking Book"
+    bl_category = "Plasma"
+    bl_description = "Adds a new Plasma Linking Book"
+    bl_options = {"REGISTER", "UNDO"}
+
+    anim_offsets = {
+        "Stand": (0.2, 1.7, -3.4),
+        "Kneel": (0.8, 2.5, -1.8),
+    }
+
+    # Allows user to specify their own name stem
+    panel_name = bpy.props.StringProperty(name="Name",
+                                         description="Linking Book name stem",
+                                         default="LinkingBook",
+                                         options=set())
+    link_anim_type = bpy.props.EnumProperty(name="Link Animation",
+                                            description="Type of Linking Animation to use",
+                                            items=[("Stand", "Standing", "The avatar steps up to the book and places their hand on the panel"),
+                                                   ("Kneel", "Kneeling", "The avatar kneels in front of the book and places their hand on the panel"),],
+                                            default="Stand",
+                                            options=set())
+
+    def draw(self, context):
+        layout = self.layout
+        space = bpy.context.space_data
+
+        if not space.local_view:
+            box = layout.box()
+            box.label("Linking Book Name:")
+            row = box.row()
+            row.alert = not self.panel_name
+            row.prop(self, "panel_name", text="")
+            box.label("Options:")
+            row = box.row()
+            row.prop(self, "link_anim_type", text="Type")
+        else:
+            row = layout.row()
+            row.label("Warning: Operator does not work in local view mode", icon="ERROR")
+
+    def execute(self, context):
+        if context.mode == "OBJECT":
+            self.create_linkingbook_objects()
+        else:
+            self.report({"WARNING"}, "Linking Book creation only valid in Object mode")
```

Comment:
Add this check to the operator `poll()` for less error checking logic.

Reply:
Done.  However, it now fails ungracefully if one tries to Add while in Local View mode (which should be disabled).  I'm investigating the correct way to filter this.

Reply:
Looks like a good solution is to toggle out of Local View mode if we're in it before executing the function here.

```
if context.space_data.local_view:
    bpy.ops.view3d.localview()
```